### ISSUE TITLE
Standardize LastWriteTime value for nonexistent files

### ISSUE THREAD

**Issue by ccastanedaucf** [dotnet/msbuild#3699]
MSBuild has mixed usage on whether ```LastWriteTime``` of a nonexistent file is represented by ```DateTime.MinValue``` or ```DateTime.FromFIleTimeUtc(0)``` (.NET API's choice of value).

This leads to boilerplate code converting from one format to the other:
```csharp
DateTime lastWriteTime = File.GetLastWriteTimeUtc(fullPath);
bool fileExists = lastWriteTime != DateTime.FromFileTimeUtc(0);

fileModifiedTime = fileExists ? lastWriteTime : DateTime.MinValue;
```

Unclear results requiring to dig through code (what if ```componentTimestamp``` was populated via ```NativeMethodsShared``` returning ```DateTime.MinValue```?):
```csharp
if (DateTime.Compare(File.GetLastWriteTime(ReferenceInfo.strippedTypeLibPath), componentTimestamp) != 0)
{
    // blah blah
}
```

Or having to check both values:
```csharp
private bool IsExistingFileTime(DateTime lastModified)
{
    return lastModified != DateTime.MinValue && lastModified != DateTime.FromFileTimeUtc(0);
}
```

Which is unavoidable in some spots due to delegates (is this ```NativeMethodsShared.GetLastWriteFileUtcTime()``` or ```File.GetLastWriteTimeUtc()```?):
```csharp
internal delegate DateTime GetLastWriteTime(string path);
```

I think the ideal solution is to follow what's already returned by .NET APIs and use a constant for ```DateTime.FromFileTimeUtc(0)```.

**Comment by rainersigwald** [comment]
Agreed: we should unify, and we should unify on "what corefx does".
